devices/zyxel-nwa50ax/default.nix

```diff
@@ -0,0 +1,300 @@
+      };
+    in {
+      imports = [ 
+        ../../modules/arch/mipsel.nix
+        ../../modules/ubinize.nix
+      ];
+
+      filesystem = dir {
+        lib = dir {
+          firmware = dir {
+            mediatek = dir {
+              "mt7915_wa.bin" = symlink wlan_firmware;
+              "mt7915_wm.bin" = symlink wlan_firmware';
+              "mt7915_rom_patch.bin" = symlink wlan_firmware'';
+            };
+          };
+        };
+      };
+
+      rootfsType = "ubifs";
+      hardware = {
+        # Taken from OpenWRT
+        # root@OpenWrt:/# ubinfo /dev/ubi0
+        # ubi0
+        # Volumes count:                           2
+        # Logical eraseblock size:                 126976 bytes, 124.0 KiB
+        # Total amount of logical eraseblocks:     256 (32505856 bytes, 31.0 MiB)
+        # Amount of available logical eraseblocks: 0 (0 bytes)
+        # Maximum count of volumes                 128
+        # Count of bad physical eraseblocks:       0
+        # Count of reserved physical eraseblocks:  19
+        # Current maximum erase counter value:     2
+        # Minimum input/output unit size:          2048 bytes
+        # Character device major/minor:            250:0
+        # Present volumes:                         0, 1
+        ubi = {
+          minIOSize = "2048";
+          logicalEraseBlockSize = "126976";
+          physicalEraseBlockSize = "128KiB";
+          maxLEBcount = "256";
+          includeBootImage = false;
+        };
+
+        defaultOutput = "ubimage";
+        loadAddress = lim.parseInt "0x80001000";
+        entryPoint = lim.parseInt "0x80001000";
+        # Aligned on 2kb.
+        alignment = 2048;
+
+        flash = {
+          address = lim.parseInt "0xbc050000";
+          size = lim.parseInt "0xfb0000";
+          eraseBlockSize = 65536;
+        };
+        rootDevice = "/dev/ubiblock0_1";
+
+        dts = {
+          src = "${openwrt.src}/target/linux/ramips/dts/mt7621_zyxel_nwa50ax.dts";
+          includes = [
+            "${openwrt.src}/target/linux/ramips/dts"
+          ];
+        };
+        networkInterfaces =
+          let
+            inherit (config.system.service.network) link;
+          in {
+            eth = link.build { ifname = "eth0"; };
+            lan = link.build { ifname = "lan"; };
+            wlan0 = link.build {
+              ifname = "wlan0";
+              dependencies = [ mac80211 ];
+            };
+            wlan1 = link.build {
+              ifname = "wlan1";
+              dependencies = [ mac80211 ];
+            };
+          };
+      };
+      boot = {
+        imageFormat = "fit";
+        # /dev/mtd5 is the rootfs of 32MB.
+        commandLine = [ "ubi.mtd=ubi" "ubi.block=0,1" ];
+        tftp = {
+          # 20MB seems to give enough room to uncompress the kernel
+          # without anything getting trodden on. 10MB was too small
+          loadAddress = lim.parseInt "0x2000000";
+          useMemmap = true;
```

Comment:
https://gti.telent.net/dan/liminix/commit/231c2cef0323d23ea61b9780442b19ba140a985a I think I fixed the bug that was causing it to need this